### PR TITLE
Feat/rush init project add template json

### PR DESCRIPTION
**Pull request by nanaSun**
### Basic Checks

Have you run `rush change` for this change?

- [x] Yes
- [ ] No

If **No**, please run `rush change` before, this is necessary.

If adding a **new feature**, the PR's description includes:

- [ ] Reason for adding this feature
- [x] How to use
- [x] A basic example

**Does this PR introduce a breaking change?** (check one)

- [x] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

### Summary

* Adding a new feature, you can create `template.json` to manage templates. 
* Assign  `unscopedPackageName` after inquirer.

### CODE REVIEW

File: rush-plugins/rush-init-project-plugin/src/logic/template.schema.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      "description": "A list of templates managed by this tool.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "displayName": {
+            "description": "The display name of the template.",
+            "type": "string"
+          },
+          "folderName": {
```
Comment (chengcyber) [inline rush-plugins/rush-init-project-plugin/src/logic/template.schema.json:21]: templateFolder

File: rush-plugins/rush-init-project-plugin/src/logic/template.schema.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "description": "The path to the template folder relative to the template config file.",
+            "type": "string"
+          },
+          "description": {
```
Comment (chengcyber) [inline rush-plugins/rush-init-project-plugin/src/logic/template.schema.json:25]: What's the usage of `description`, `tags`, `author` ?

File: rush-plugins/rush-init-project-plugin/src/logic/template.schema.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+    "templates": {
+      "description": "A list of templates managed by this tool.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "displayName": {
```
Comment (chengcyber) [inline rush-plugins/rush-init-project-plugin/src/logic/template.schema.json:17]: If displayName is moved to `template.json`, we'd better remote it in `init.config`. That follows our further direction.

**Review by chengcyber — CHANGES_REQUESTED** [review]

File: rush-plugins/rush-init-project-plugin/src/logic/template.schema.json (hunk 2 above)
Comment (nanaSun) [reply rush-plugins/rush-init-project-plugin/src/logic/template.schema.json:25]: `description`, `tags`, `author` might be used in the future, I will remove it now.

**Review by chengcyber — APPROVED** [review]

**Pull request merged**